### PR TITLE
Update LS deployment docs with Alvin's content

### PR DESCRIPTION
**Pull request by dedemorton**
The PR contains the deployment content that Alvin created. This is a complete replacement of the original text.

@acchen97 I've made a few minor changes (see the comments I've added to the source asciidoc). I've also added one or two questions.

ToDo's (after review): 
- [x] Make sure that none of the deleted IDs are referenced in cross-doc links
- [ ] After merging into master, cherry-pick the changes manually into 5.4 and 5.x. Because there are changes in the index.asciidoc, using Jarvis to merge the changes might mess up the paths.

### CODE REVIEW

File: docs/static/deploying.asciidoc [hunk 1]
```diff
@@ -1,146 +1,264 @@
...
+gets scalably, durably, and securely transported to Elasticsearch is extremely
+important, especially for mission critical environments.
+
+The goal of this document is to highlight the most common architecture patterns
+for Logstash and how to effectively scale as your deployment grows. The focus
+will be around the operational log, metrics, and security analytics use cases
+because they tend to require larger scale deployments. The deploying and scaling
+recommendations provided here may vary based on your own requirements.
+
+//ALVIN: Changed "The focus will be around the operational log, metrics, and security analytics use cases as they tend to run at formidable ingestion scale" to what you see here.
```
Comment (acchen97) [inline docs/static/deploying.asciidoc]: ++ works for me.

File: docs/static/deploying.asciidoc [hunk 2]
```diff
@@ -1,146 +1,264 @@
...
+* Wire encryption is recommended for both the transport from
+{filebeat}configuring-ssl-logstash.html[Beats to Logstash] and from 
+{xpack}/logstash.html[Logstash to Elasticsearch].
+* There’s a wealth of security options when communicating with Elasticsearch
+including basic authentication, TLS, PKI, LDAP, AD, and other custom realms.
+To enable Elasticsearch security, consult the
+{xpack}/xpack-security.html[X-Pack documentation].
 
-image::static/images/deploy_4.png[]
+//ALVIN: I've changed the links here a bit from what you had in your doc. I thought it was potentially confusing to have  links pointing to configuration options in the Beats and LS doc because the reference topics don't give you much context, so I've pointed to a conceptual topic in the Beats doc that covers the Beats & Logstash config.
```
Comment (acchen97) [inline docs/static/deploying.asciidoc]: good point, the links you have look good.

File: docs/static/deploying.asciidoc [hunk 3]
```diff
@@ -1,146 +1,264 @@
...
-your data source machines and the Logstash cluster. The load balancer handles the individual connections to the
-Logstash instances to ensure continuity of data ingestion and processing even when an individual instance is unavailable.
+The TCP, UDP, and HTTP protocols are common ways to feed data into Logstash.
+Logstash can expose endpoint listeners with the respective
+<<plugins-inputs-tcp,TCP>>, <<plugins-inputs-udp,UDP>>, and
+<<plugins-inputs-http,HTTP>> input plugins. The data sources enumerated below
+are typically ingested through one of these three protocols.
 
-image::static/images/deploy_6.png[]
+//ALVIN: Not sure what we are referring to when we say "the data sources enumerated below"
```
Comment (acchen97) [inline docs/static/deploying.asciidoc]: network/security data, syslog servers, iot data all come through either the TCP, UDP, or HTTP protocols.

File: docs/static/deploying.asciidoc [hunk 4]
```diff
@@ -1,146 +1,264 @@
...
-Any of these layers can be scaled by adding computing resources. Examine the performance of these components regularly
-as your use case evolves and add resources as needed. When Logstash routinely throttles incoming events, consider
-adding storage for your message broker. Alternately, increase the Elasticsearch cluster's rate of data consumption by
-adding more Logstash indexing instances.
+Infrastructure and application metrics can be collected with
+https://www.elastic.co/products/beats/metricbeat[Metricbeat], but they can also
+send webhooks to a Logstash HTTP input or have metrics polled from an HTTP
+endpoint with the <<plugins-inputs-http_poller,HTTP poller input plugin>>.
+
+//ALVIN: Unclear above as to what "they" refers back to. Should "they" be "you"?
```
Comment (acchen97) [inline docs/static/deploying.asciidoc]: "they" means the applications themselves

**Comment by acchen97** [comment]
Looks good overall, I'm good with your edits.

File: docs/static/deploying.asciidoc (hunk 3 above)
Comment (dedemorton) [inline docs/static/deploying.asciidoc]: the statement is a bit broad, but it's OK, so I'll leave it as-is.

File: docs/static/deploying.asciidoc (hunk 4 above)
Comment (dedemorton) [inline docs/static/deploying.asciidoc]: ok, changed to "applications"

**Pull request closed without merging**